server/app/mutations/grid_services/restart.rb

```diff
@@ -1,35 +1,27 @@
 module GridServices
   class Restart < Mutations::Command
-    include AsyncHelper
-
     required do
       model :grid_service
     end
 
     def execute
-      async_thread do
-        self.restart_service_instances
-      end
+      self.restart_service_instances
+    rescue => exc
+      add_error(:restart, :error, exc.message)
```

Comment:
Does this change behaviour? (or does it matter)

Reply:
Yes, in that `kontena service restart` now behaves the same as `kontena service start/stop`... however, because of the `async_thread`, the end result from the CLI perspective should be more or less the same... the CLI/API wasn't actually waiting for the services to restart, or reporting any errors.

See #3277 for how all three of those should eventually be changed to run as a deploy instead.

Reply:
There's also the `grid_service.set_state('restarting')` / `grid_service.set_state('running')` that's now gone, but I think that's more of a pre-1.2 remnant than anything else...